packages/playwright/src/reporters/merge.ts

```diff
@@ -508,3 +519,39 @@ class JsonEventPatchers {
     }
   }
 }
+
+class BlobModernizer {
+  modernize(fromVersion: number, events: JsonEvent[]): JsonEvent[] {
+    const result = [];
+    for (const event of events)
+      result.push(...this._modernize(fromVersion, event));
+    return result;
+  }
+
+  private _modernize(fromVersion: number, event: JsonEvent): JsonEvent[] {
+    let events = [event];
+    for (let version = fromVersion; version < currentBlobReportVersion; ++version)
+      events = (this as any)[`_modernize_${version}_to_${version + 1}`].call(this, events);
+    return events;
+  }
+
+  _modernize_1_to_2(events: JsonEvent[]): JsonEvent[] {
+    return events.map(event => {
+      if (event.method === 'onProject') {
+        const modernizeSuite = (suite: blobV1.JsonSuite) => {
+          for (const child of suite.suites)
+            modernizeSuite(child);
+          (suite as any).entries = [...suite.suites, ...suite.tests];
```

Comment:
Something like this can help getting rid of any. Otherwise there is not much point in strictness.

```js
const { suites, tests, ...remainder } = suite;
return { entries: [...suites, ...tests], ...remainder };
```

Reply:
Done.